### ISSUE TITLE
SCA: Allow customer to come back to authenticate recurring payment

### ISSUE THREAD

**Issue by v18** [woocommerce/woocommerce-gateway-stripe#937]
Relies on #957

When a recurring payment (for a subscription) requires extra authentication, we need to:

- send an email to the customer to ask them to authenticate the payment via a link
- allow customer to authenticate their payment info at the link provided in the email
- let store merchants customize the email

The flow should be similar to the credit card expiration flow.

This needs to work for:
- Subscriptions
- Pre-orders
- custom subscriptions - needs to make sense in terms of order status, payment status, etc
- so we may or may not need a logged-out link

Note, that this flow is needed even if payment details are authenticated the first time a customer enters them. This is because even though they were already authenticated, the issuing bank may explicitly ask for re-authentication when the are re-used.

**Comment by tommyshellberg** [comment]
If I'm not mistaken this could be done through the Setup Intents API:

https://stripe.com/docs/api/setup_intents

It would then use the PaymentIntents API to charge when necessary, marking it as a "merchant-initiated transaction" to hopefully avoid needing to reauthenticate.

I'm speaking specifically here about Pre-Orders(previously tracked through #802 )

>Crowdfunding platforms typically capture payments more than seven days after authorization. Each campaign lasts for a set length of time, and payments are captured when a campaign is successful. If your business has a similar payment flow, **you can authenticate with 3D Secure when customers pledge to support a campaign**, and then authorize and capture when the campaign ends successfully.

At the moment no 3DS auth is performed when checking out if users are charged later.

![https://cld.wthms.co/3PZpB2](https://cld.wthms.co/3PZpB2+)
Image link: https://cld.wthms.co/3PZpB2

![https://cld.wthms.co/8hlnpP](https://cld.wthms.co/8hlnpP+)
Image link: https://cld.wthms.co/8hlnpP

**Comment by v18** [comment]
@tommyshellberg thanks for writing up your research. We will be making sure that subsequent purchases won't need to be authenticated, like you outlined. However, even if we take all of those steps to ensure that authentication happens when a customer fills our their payment details the first time, some banks will ask for subsequent payments to be re-authenticated anyway. This is why we need the flow of getting a user back on the site to re-authenticate.

That's documented at this doc under 'Recurring payments':

https://stripe.com/docs/strong-customer-authentication/migration#step-2

It says:

> However, banks can decide to reject a request for exemption. Build a way to notify customers that they need to return to your application and complete authentication if required.

And links to this doc:

https://stripe.com/docs/payments/cards/charging-saved-cards#notify

**Comment by v18** [comment]
Some related designs: p6JqRr-2di-p2

**Comment by allendav** [comment]
@RadoslavGeorgiev - you mention in #991 that this "fixes the email part" of this issue. Would you add a checklist to the issue description above with all the parts?

**Comment by RadoslavGeorgiev** [comment]
> When a recurring payment (for a subscription) requires extra authentication, we need to:

- _send an email to the customer to ask them to authenticate the payment via a link_
Handled by #991 for both subscriptions and pre-orders.
- _allow customer to authenticate their payment info at the link provided in the email_
The link is generated by #986 and works for both subscriptions and pre-orders as well.
- _let store merchants customize the email_
PR #991 should handle this as well, considering we're talking about standard WooCommerce Emails. There are some (very) basic settings in the options of the email, but the template can also be overwritten.

This needs to work for:

- _Subscriptions:_ It does.
- _Pre-orders:_ It does.
- _custom subscriptions - needs to make sense in terms of order status, payment status, etc_: From my perspective what we're doing already makes sense in terms of order status, and payment status is not a thing in WooCommerce yet.
- _so we may or may not need a logged-out link_: I tested this, and indeed the link does not work when logged out. I opened #1037 to handle this.
